lib/dal/src/attribute/value/subscription.rs

```diff
@@ -23,4 +27,15 @@ impl ValueSubscription {
     ) -> AttributeValueResult<Option<AttributeValueId>> {
         self.path.resolve(ctx, self.attribute_value_id).await
     }
+
+    /// Find or create the attribute value a subscription points to
+    pub async fn vivify(&self, ctx: &DalContext) -> AttributeValueResult<AttributeValueId> {
+        self.path.vivify(ctx, self.attribute_value_id).await
+    }
```

Comment:
From the doc comment, I'm not sure why this would ever be called?

If `Component A: /domain/some/destination/field` is subscribing to `Component B: /domain/data/source/field`, then `Component B: /domain/data/source/field` should only exist as an AV if something about `Component B` has determined that it should (manually set, function output from B, etc.), the fact that `Component A` is interested in the data in `Component B` should not affect the structure of `B` in any way.

Reply:
It was mainly so the tests could call it, but they really don't mean subscription here, they are using an (AV_ID, path) pair as a *reference* to something they want to set. I'll remove this and just have the caller do that directly, so there isn't confusion!